lib/core/src/recover/handlers/handle_send_swap.rs

```diff
@@ -61,25 +61,33 @@ impl SendSwapHandler {
         let mut recovered_data = Self::recover_onchain_data(&context.tx_map, &swap_id, history)?;
 
         // Recover preimage if needed
-        if let (Some(claim_tx_id), None) = (&recovered_data.claim_tx_id, &send_swap.preimage) {
-            match Self::recover_preimage(
-                context,
-                claim_tx_id.txid,
-                &swap_id,
-                context.swapper.clone(),
-            )
-            .await
+        if recovered_data.lockup_tx_id.is_some() && send_swap.preimage.is_none() {
+            let pair_limits = send_swap.get_boltz_pair()?.limits;
+            // If the swap is batched or we have a claim tx id, we can attempt
+            // to recover the preimage cooperatively. If we cannot recover it
+            // cooperatively, we can try to recover it from the claim tx.
+            if send_swap.receiver_amount_sat < pair_limits.minimal
```

Comment:
I am not sure we need to check if it is under the limit. I think botz expose the preimage on TransactionClaimPending pending state. I asked Michael.

Reply:
Removed [4208d70](https://github.com/breez/breez-sdk-liquid/pull/863/commits/4208d701b7f527440b438c8a8259259a6531550f)